### PR TITLE
SLS-518 Fix largest key implementation and implement a simple random cursor for oligarch table

### PR DESCRIPTION
**Pull request by quchenhao**
Fix the implementation of the largest key implementation. The previous implementation only implements the largest visible keys. However, largest key should return the largest key even it's not visibility.

Implement the random cursor function for oligarch table. I borrowed the main implementation from the lsm code. It is not a uniform random solution yet as it doesn't consider the key size of the ingest table and the stable table but it should work for now.

Fix a bug in oligarch search near.

### CODE REVIEW

File: src/cursor/cur_oligarch.c [hunk 1]
```diff
@@ -803,7 +803,7 @@ __coligarch_search_near(WT_CURSOR *cursor, int *exactp)
     ingest_found = ret != WT_NOTFOUND;
 
     /* If there wasn't an exact match, check the stable table as well */
-    if (ingest_cmp != 0) {
+    if (!ingest_found || ingest_cmp != 0) {
```
Comment (quchenhao) [inline src/cursor/cur_oligarch.c:806]: It is a bug in oligarch search near.

File: src/cursor/cur_file.c [hunk 2]
```diff
@@ -1097,6 +1097,54 @@ __curfile_create(WT_SESSION_IMPL *session, WT_CURSOR *owner, const char *cfg[],
     return (ret);
 }
 
+/*
+ * __wti_cursor_largest_key --
+ *     WT_CURSOR->largest_key default implementation..
+ */
+int
+__wti_cursor_largest_key(WT_CURSOR *cursor)
```
Comment (pmacko86) [inline src/cursor/cur_file.c:1105]: Why did you move this function? It's a generic implementation that's not specific to file or even btree cursors. It's still in `src/cursor/cur_std.c` on develop.

File: src/cursor/cur_oligarch.c [hunk 3]
```diff
@@ -1265,6 +1289,53 @@ __wt_coligarch_close(WT_CURSOR *cursor)
     API_END_RET(session, ret);
 }
 
+/*
+ * __coligarch_next_random --
+ *     WT_CURSOR->next_random method for the oligarch cursor type.
+ */
+static int
+__coligarch_next_random(WT_CURSOR *cursor)
```
Comment (pmacko86) [inline src/cursor/cur_oligarch.c:1297]: This looks good for now, but it would be better if the random cursor would also consider the ingest table—not just when the stable table is empty. Could you please add a TODO?

File: src/cursor/cur_oligarch.c [hunk 4]
```diff
@@ -1170,38 +1170,62 @@ __coligarch_reserve(WT_CURSOR *cursor)
 static int
 __coligarch_largest_key(WT_CURSOR *cursor)
```
Comment (pmacko86) [inline src/cursor/cur_oligarch.c:1171]: Just curious, what was the issue with the generic implementation that just called `prev` on the oligarch cursor? If it didn't work, it probably means that we have a bug in `prev`?

File: src/cursor/cur_file.c (hunk 2 above)
Comment (quchenhao) [reply src/cursor/cur_file.c:1105]: It needs to use the WT_WITH_CHECKPOINT macro

File: src/cursor/cur_oligarch.c (hunk 4 above)
Comment (quchenhao) [reply src/cursor/cur_oligarch.c:1171]: It need to set the WT_CURSTD_KEY_ONLY on the ingest table and stable table's cursor not on the high level oligarch cursor. However, internally the coligarch_prev function may open and close those cursors so setting the flags in the largest_key function doesn't work.

**Review by korteland — APPROVED** [review]
LGTM!
